driver-core/src/main/java/com/datastax/driver/core/RequestHandler.java

```diff
@@ -150,13 +150,6 @@ private void cancelPendingExecutions(SpeculativeExecution ignore) {
             execution.cancel();
     }
 
-    private void logError(InetSocketAddress address, Throwable exception) {
```

Comment:
Moved inside SpeculativeExecution to have the spec exec id logged as well. Fixed the check-then-act idiom.